src/components/RegionLabel.js

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { useCloudRegions } from '../hooks/apis/useCloudRegions';
+import { useMemo } from 'react';
+import { AWS_PROVIDER } from '../utils/cloudProvider';
+import { getRegionDisplayName } from '../utils/region';
+
+export default function RegionLabel({ id }) {
+  const { data: regionList } = useCloudRegions({ provider: AWS_PROVIDER });
+  const regions = useMemo(() => regionList?.items || [], [regionList]);
+  const region = useMemo(() => regions.find((r) => r.id === id), [regions, id]);
```

Comment:
nit: The first of these memo-izations are a little overkill, and the optimization the build step runs in a case like this could actually make a sub-millisecond dereferencing optimization and make it take a little longer.

It's fine either way, because we're talking about a very finite number in the table.

Or you could just combine both statements, like this:
```suggestion
  const region = useMemo(() => regionList?.items?.find((r) => r.id === id), [regionList, id]);
```

The first optional chaining, `?.`, will short-circuit to `undefined` if `items` is not present. The second optional chaining will short-circuit if it's not an array. Either way, the `getRegionDisplayName` should be set up to handle an undefined `region` argument gracefully.